Clother/View Models/ClothesViewModel.swift

```diff
@@ -26,59 +26,145 @@ class ClothesViewModel {
+            }
+        } else {
+            randomTop = nil
+        }
+        
+        if !colorApplied {
+            var tempFilteredBottoms: [ClothingItem] = []
+            if !fabricApplied {
+                tempFilteredBottoms = filteredBottoms.filter({ $0.color == color && $0.fabric == fabric })
+            } else {
+                tempFilteredBottoms = filteredBottoms.filter({ $0.color == color })
+            }
+            filteredBottoms = tempFilteredBottoms
+        }
+        
+        if let bottom = filteredBottoms.randomElement() {
+            randomBottom = bottom
+        } else {
+            randomBottom = nil
+        }
+        
+        if let shoe = filteredShoes.randomElement() {
+            randomShoe = shoe
+        } else {
+            randomShoe = nil
+        }
+        
+        for _ in 0..<min(3, filteredAccessories.count) {
+            let randomAccessory = filteredAccessories.randomElement()!
+            randomAccessories.append(randomAccessory)
+        }
+        return (randomTop, randomBottom, randomShoe, randomAccessories)
     }
     
     
     // TODO: Saves the outfit to the necessary collections.
     func saveOutfit(item: OutfitItem) {
-        
+        userOutfits.append(item)
+        outfitsOnDisplay = userOutfits
+    }
+    // TODO: Saves the outfit to the necessary collections.
+    func saveClothing(item: ClothingItem) {
+        userClothes.append(item)
+        clothesOnDisplay = userClothes
     }
-    
     // TODO: Manipulate the closet items on display such that only items with the provided input in their name are visible.
     func search(input: String) {
-        
+        clothesOnDisplay = userClothes.filter {$0.name.lowercased().contains(input.lowercased())}
     }
     
     // TODO: Manipulate the clothing items on display such that only items with the provided filters are visible.
     /*
      If a given parameter is nil, it means that no filter was applied for it.
      */
     func filterClothes(by filter: Filter) {
-        
+        clothesOnDisplay = userClothes.filter { clothingItem in
+            let typeMatch = filter.type == nil || clothingItem.type == filter.type
+            let sizeMatch = filter.size == nil || clothingItem.size == filter.size
+            let colorMatch = filter.color == nil || clothingItem.color == filter.color
+            let weatherMatch = filter.weather == nil || clothingItem.weather == filter.weather
+            let fabricMatch = filter.fabric == nil || clothingItem.fabric == filter.fabric
+            
+            return typeMatch && sizeMatch && colorMatch && weatherMatch && fabricMatch
+        }
     }
     
     // TODO: Manipulate the outfit items on display such that only items with the provided filters are visible.
     /*
      If a given parameter is nil, it means that no filter was applied for it.
      */
     func filterOutfits(topFilter: Filter, bottomFilter: Filter, shoesFilter: Filter) {
```

Comment:
Filtering doesn't take into accounts cases where Filter object properties are nil (ex: color, type). Fix pls.

Reply:
Does it not already do that because if nil is passed it just sets it to true

Reply:
Nope, it doesn't work that way. Nil means no value. You can't say it equals something with real value.